### PR TITLE
VideoPress: reorganize and polish hooks to sync video Data

### PR DESCRIPTION
**Pull request by retrofox**
# branched off from https://github.com/Automattic/jetpack/pull/27192

This PR reorganizes the hooks' files to synchronize the video data in the block editor context and polishes the process.

Fixes #

#### Changes proposed in this Pull Request:

* VideoPress: reorganize and polish hooks to sync video Data

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Testing instructions:

Since it's a refactoring process, I suggest functionality tests:

### Test sync initial video data

* Go to block editor
* Create a Video block
* Open the sidebar
* Confirm the browser performs only one request to get the video data

Screen Shot 2022-11-02 at 05 29 49

* Change title and/or description
* Confirm the data is synced by taking a look at the async requests

Screen Shot 2022-11-02 at 05 31 30

Above, the client performs a POST request to update the video data title, hitting the REST API endpoint.

* Go to the VideoPress dashboard
* Edit the video details via the video page

image

Screen Shot 2022-11-02 at 05 34 24

* Go back to block editor
* hard-refresh
* Confirm the data is synced 
Screen Shot 2022-11-02 at 05 35 22

### Test sync chapters

* Change video chapters from the description text area
* Confirm the whole flow

https://user-images.githubusercontent.com/77539/199480684-22c40346-09e2-4529-a795-802ed5b7ef94.mov

### CODE REVIEW

File: projects/packages/videopress/src/client/block-editor/hooks/use-video-data-update/Readme.md [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
+React custom hook to keep block attributes in sync with the video data.
+The hook will keep the initial state of the attributes
+and update the data when it detects changes after the post saves.
 
 ```jsx
 import { useSyncMedia } from './use-video-item-update';
 
-export default function VideoItemComponent( { attributes, setAttributes } ) {
-	useSyncMedia( attributes );
+export default function VideoItemComponent( { setAttributes } ) {
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/block-editor/hooks/use-video-data-update/Readme.md:32]: `attributes` is used in the next line

**Review by dhasilva — APPROVED** [review]
LGTM

**Comment by lhkowalski** [comment]
I could confirm that:

- only one request for the video data was made
- meta data is updated from the block
- the meta data is up to date on the media details page, inside the VideoPress library
- changing the data on the VideoPress library also updated it on the details panel for the video
- updating chapters on the details panel and saving the post will trigger a player reload with the updated list of chapters
- updating chapters on the video details page will not, by itself, generate new chapters on the player (as described by Douglas on Slack)

**Review by lhkowalski — APPROVED** [review]
Changes looks good!

**Pull request merged**